docs/azure/sdk/unit-testing-mocking.md

```diff
@@ -0,0 +1,420 @@
+{
+    NotBefore = DateTimeOffset.Now
+};
+```
+
+To create instances of output models, a model factory is used. Most Azure SDK client libraries provide a static model factory class that ends in `ModelFactory` and contains a set of static methods to initialize the library's output model types.
+
+```csharp
+KeyVaultSecret keyVaultSecret = SecretModelFactory.KeyVaultSecret(
+    new SecretProperties("secret"), "secretValue");
+```
+
+> [!NOTE]
+> Some input models have read-only properties that are only populated when the model is returned by the service. In this case, a model factory method will be available that allows setting these properties. See <xref:Azure.Security.KeyVault.Secrets.SecretModelFactory>.
+
+```csharp
+// CreatedOn is a read-only property and can only be 
+// set via a model factory SecretProperties.
+secretPropertiesWithCreatedOn = SecretModelFactory.SecretProperties(
+    name: "secret", createdOn: DateTimeOffset.Now);
+```
+
+## Explore response types
+
+The <xref:Azure.Response> class is an abstract class that represents an HTTP response and is a part of almost all types returned by service client methods. You can create test `Response` instances using either the Moq library or standard C# inheritence.
+
+## [Moq](#tab/moq)
+
+The Moq library provides concise functionality for setting up mock responses:
+
+```csharp
+Mock responseMock = new Mock();
+responseMock.SetupGet(r => r.Status).Returns(200);
+
+Response response = responseMock.Object;
+```
+
+To create an instance of `Response` without defining any behaviors:
+
+```csharp
+Response response = Mock.Of<Response>();
+```
+
+## [C#](#tab/csharp)
+
+The `Response` class is abstract, which means there are a lot of members to override. Consider using the Moq library to streamline your approach.
+
+```csharp
+public class TestResponse : Response
+{
+    protected override bool TryGetHeader(string name, out string value)
+        => throw new NotImplementedException();
+    protected override bool TryGetHeaderValues(string name, out IEnumerable<string> values)
+        => throw new NotImplementedException();
+         
+    protected override bool ContainsHeader(string name) => throw new NotImplementedException();
+    protected override IEnumerable<HttpHeader> EnumerateHeaders() 
+        => throw new NotImplementedException();
+
+    public override int Status => throw new NotImplementedException();
+    public override string ReasonPhrase => throw new NotImplementedException();
+    public override Stream ContentStream => throw new NotImplementedException();
+    public override string ClientRequestId => throw new NotImplementedException();
+
+    public override void Dispose() => throw new NotImplementedException();
+}
+```
+
+---
+
+Some services also support using the <xref:Azure.Response%601> type, which is a class that contains a model and the HTTP response that returned it. To create a test instance of `Response<T>`, use the static `Response.FromValue` method:
```

Comment:
@christothes do we need a section on NullableResponse?

Reply:
I would just mention it as a possible return type, but the same principals apply from a mocking perspective.

Reply:
I thought we weren't supposed to return that from any APIs.

Reply:
It should be returned from operations where null is a possible expected success result. Example [here](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/tables/Azure.Data.Tables/api/Azure.Data.Tables.netstandard2.0.cs#L40-L41)

Reply:
We can always add that later if necessary. Resolving for now.